### PR TITLE
Bump CDAP and Hadoop dependencies

### PR DESCRIPTION
**Pull request by dli357**
Bump CDAP version to 6.8, hydrator-plugins to 2.10.0, and hadoop to 2.10.2.

Moved config builder classes to src to allow e2e tests to compile, and bumped e2e test version to make it pass.

Tested with unit tests and e2e tests.

### CODE REVIEW

**Review by DJSagarAhire — APPROVED** [review]
LGTM

File: pom.xml [hunk 1]
```diff
@@ -58,11 +58,11 @@
...
     <avro.version>1.7.7</avro.version>
-    <cdap.version>6.8.0-SNAPSHOT</cdap.version>
+    <cdap.version>6.8.0</cdap.version>
     <junit.version>4.12</junit.version>
     <commons.io.version>2.6</commons.io.version>
     <commons.lang.version>2.6</commons.lang.version>
-    <hadoop.version>2.3.0</hadoop.version>
+    <hadoop.version>2.10.2</hadoop.version>
     <spark2.version>2.1.3</spark2.version>
     <hydrator.version>2.10.0-SNAPSHOT</hydrator.version>
```
Comment (vsethi09) [inline pom.xml]: Should this be `2.10.0` instead of `2.10.0-SNAPSHOT`?

**Review by vsethi09 — APPROVED** [review]
Left one comment, apart from that LGTM.

File: pom.xml (hunk 1 above)
Comment (dli357) [reply pom.xml]: Fixed, thanks!

**Pull request merged**